From c77e13b9b389df893f6b1e7ae8e488d7ead8ce5a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20Cr=C3=AAte?= <olivier.crete@collabora.com>
Date: Mon, 26 Sep 2016 20:06:54 -0400
Subject: [PATCH] avcodecmap: Enable 24 bit WMA Lossless decoding

This now works with not so recent ffmpeg.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-libav/-/merge_requests/88>
---
 docs/gst_plugins_cache.json | 2 +-
 ext/libav/gstavcodecmap.c   | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/docs/gst_plugins_cache.json b/docs/gst_plugins_cache.json
index afac3527..016fbca0 100644
--- a/docs/gst_plugins_cache.json
+++ b/docs/gst_plugins_cache.json
@@ -22978,7 +22978,7 @@
                 "long-name": "libav Windows Media Audio Lossless decoder",
                 "pad-templates": {
                     "sink": {
-                        "caps": "audio/x-wma:\n     wmaversion: 4\n          depth: 16\n",
+                        "caps": "audio/x-wma:\n     wmaversion: 4\n",
                         "direction": "sink",
                         "presence": "always"
                     },
diff --git a/ext/libav/gstavcodecmap.c b/ext/libav/gstavcodecmap.c
index 16f1aba0..46885303 100644
--- a/ext/libav/gstavcodecmap.c
+++ b/ext/libav/gstavcodecmap.c
@@ -1220,10 +1220,9 @@ gst_ffmpeg_codecid_to_caps (enum AVCodecID codec_id,
     }
     case AV_CODEC_ID_WMALOSSLESS:
     {
-      /* libav only supports a depth of 16 at the moment */
       caps =
           gst_ff_aud_caps_new (context, NULL, codec_id, encode, "audio/x-wma",
-          "wmaversion", G_TYPE_INT, 4, "depth", G_TYPE_INT, 16, NULL);
+          "wmaversion", G_TYPE_INT, 4, NULL);
       break;
     }
     case AV_CODEC_ID_WMAVOICE:
-- 
GitLab

